lib/porky_lib/file_service.rb

```diff
@@ -91,6 +91,28 @@ def overwrite_file(file, file_key, bucket_name, key_id, options = {})
     tempfile.unlink
   end
 
+  def presigned_post_url(bucket_name, options = {})
+    file_name = options[:file_name] || SecureRandom.uuid
+    obj = s3.bucket(bucket_name).object(file_name)
+
+    presigned_url = obj.presigned_url(:put,
+                                      server_side_encryption: 'aws:kms',
```

Comment:
I don't think you need this anymore, isn't the bucket doing server side encrypting anyway? We needed this when we also specified the encryption key